### PR TITLE
Fix/vetrina login view

### LINKED ISSUE #342: Handle invalid credentials
Currently we are rendering the error view when user inserts invalid credentials.

We should rather show similar error message what we have in the `User` component:
Screenshot_2020-04-28 Mitt Konto

### PR DESCRIPTION
**Pull request by ae-mo**
Closes #342.

Contains a refactoring of `Vetrina` module too.

### CODE REVIEW

File: packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs [hunk 1]
```diff
@@ -40,14 +40,15 @@ type State =
   }
 
 type Props =
-  { accountStatus :: AccountStatus
-  , products :: Array Product
-  , mkPurchaseWithNewAccount :: NewAccountForm -> Effect Unit
+  { accountStatus                 :: AccountStatus
+  , products                      :: Array Product
+  , errorMessage                  :: JSX
```
Comment (vapaj) [inline packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs:45]: Let's change this to `String` and let `NewPurchase` decide what `JSX` to render.

File: packages/vetrina/src/Vetrina/Vetrina.purs [hunk 2]
```diff
@@ -85,11 +85,9 @@ data PurchaseState
   = NewPurchase
   | CapturePayment PaymentTerminalUrl
   | ProcessPayment
-  | PurchaseFailed
+  | PurchaseFailed (Maybe OrderFailure)
```
Comment (vapaj) [inline packages/vetrina/src/Vetrina/Vetrina.purs:88]: This should be `PurchaseFailed OrderFailure`, as we should always know what error happened.

File: packages/vetrina/src/Vetrina/Vetrina.purs [hunk 3]
```diff
@@ -98,6 +96,7 @@ data OrderFailure
   | AuthenticationError
   | ServerError
   | UnrecognizedError String
+  | UnexpectedError
```
Comment (vapaj) [inline packages/vetrina/src/Vetrina/Vetrina.purs:99]: Let's either have `UnrecognizedError String` or `UnexpectedError`, but not both.

File: packages/vetrina/src/Vetrina/Vetrina.purs [hunk 4]
```diff
@@ -166,10 +165,10 @@ didUpdate self _ = Aff.launchAff_ $ stopOrderPollerOnCompletedState self
 stopOrderPollerOnCompletedState :: Self -> Aff Unit
 stopOrderPollerOnCompletedState self =
   case self.state.purchaseState of
-    PurchaseFailed      -> killOrderPoller self.state
-    PurchaseCompleted _ -> killOrderPoller self.state
-    NewPurchase         -> killOrderPoller self.state
-    _                   -> pure unit
+    PurchaseFailed Nothing -> killOrderPoller self.state
```
Comment (vapaj) [inline packages/vetrina/src/Vetrina/Vetrina.purs:168]: We need to stop the order polling if any `PurchaseFailed` happens.

File: packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs (hunk 1 above)
Comment (ae-mo) [reply packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs:45]: But this way `NewPurchase` needs to know what kind of order failure was there correct? At least if I understood what you mean by let `NewPurchase` decide what `JSX` to render.

File: packages/vetrina/src/Vetrina/Vetrina.purs (hunk 2 above)
Comment (ae-mo) [reply packages/vetrina/src/Vetrina/Vetrina.purs:88]: What error would you return here?

https://github.com/KSF-Media/affresco/blob/master/packages/vetrina/src/Vetrina/Vetrina.purs#L205

File: packages/vetrina/src/Vetrina/Vetrina.purs (hunk 3 above)
Comment (ae-mo) [reply packages/vetrina/src/Vetrina/Vetrina.purs:99]: Semantically they are slightly different, what would be the criterion to decide? Maybe what's the more general? In that case I would go with `UnexpectedError`.

File: packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs (hunk 1 above)
Comment (vapaj) [reply packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs:45]: If we are always showing the error above the input field, it's enough to know what the message is, right?

So the thing in `Vetrina` becomes:
```purescript
orderErrorMessage :: OrderFailure -> String
orderErrorMessage failure =
  case failure of
    AuthenticationError -> "Kombinationen av e-postadress och lösenord finns inte"
    _ -> DOM.text "Något gick fel. Vänligen försök om en stund igen."
```

File: packages/vetrina/src/Vetrina/Vetrina.purs (hunk 3 above)
Comment (vapaj) [reply packages/vetrina/src/Vetrina/Vetrina.purs:99]: Yeah, maybe with the message as well: `UnexpectedError String`. What do you think?

File: packages/vetrina/src/Vetrina/Vetrina.purs (hunk 2 above)
Comment (vapaj) [reply packages/vetrina/src/Vetrina/Vetrina.purs:88]: Depends on what the `OrderStatusFailReason` is (https://github.com/KSF-Media/affresco/blob/master/packages/components/src/Bottega.purs#L82-L89).

I'm currently implementing the `SubscriptionExistsError` one, but I think in other cases we can default to `ServerError` or `UnexpectedError` even though they're not exactly true (It probably would not be beneficial to reflect all the `OrderStatusFailReason` values to the user interface)

File: packages/vetrina/src/Vetrina/Vetrina.purs (hunk 3 above)
Comment (ae-mo) [reply packages/vetrina/src/Vetrina/Vetrina.purs:99]: I agree 👍

File: packages/vetrina/src/Vetrina/Vetrina.purs [hunk 5]
```diff
@@ -238,10 +236,33 @@ render self = vetrinaContainer self $
...
     CapturePayment url -> netsTerminalIframe url
     ProcessPayment -> Spinner.loadingSpinner
-    PurchaseFailed ->
-      Purchase.Error.error
-        { onRetry
-        }
+    PurchaseFailed failure ->
+      case failure of
+        SubscriptionExists ->
+          DOM.div_
```
Comment (vapaj) [inline packages/vetrina/src/Vetrina/Vetrina.purs:242]: I guess this got lost in the rebase conflict mess 😄 Should be 
```purescript
Purchase.SubscriptionExists.subscriptionExists
  { onClose: self.props.onClose }
```

File: packages/vetrina/src/Vetrina/Vetrina.purs [hunk 6]
```diff
@@ -333,14 +347,14 @@ mkPurchase self@{ state: { logger } } validForm affUser = Aff.launchAff_ $ Spinn
         startOrderPoller self.setState newState order
     Left err -> do
       case err of
-        UnrecognizedError e -> liftEffect $ logger.error $ Error.orderError $ "Failed to place an order: " <> e
+        UnexpectedError e -> liftEffect $ logger.error $ Error.orderError $ "Failed to place an order: " <> e
         _ -> pure unit
 
       let errState = { user: hush eitherUser } `merge` case err of
```
Comment (vapaj) [inline packages/vetrina/src/Vetrina/Vetrina.purs:353]: I think we lost the case for `AuthenticationError` here

**Pull request merged**